packages/gasket-plugin-middleware/lib/internal.d.ts

```diff
@@ -72,17 +72,78 @@ export function executeMiddlewareLifecycle(
  */
 export function attachLogEnhancer(req: ExpressRequest): void;
 export function isValidMiddleware(middleware: Function | Function[]): boolean;
+
+/**
+ * Checks whether a given value is a middleware object with a `handler` function.
+ *
+ * @param {any} value - The value to check
+ * @returns {value is { handler: Handler; paths?: string | RegExp | Array<string | RegExp> }} True if the value is a middleware object
+ */
+export function isMiddlewareObject(value: any): value is { handler: Handler; paths?: string | RegExp | Array<string | RegExp> };
+
+/**
+ * Normalizes a middleware entry into a consistent `{ handler, paths }` format.
+ * Accepts either a function, an object with a `handler`, or an invalid entry.
+ *
+ * @param {any} entry - The middleware entry to normalize
+ * @returns {{ handler: Handler | undefined, paths?: string | RegExp | Array<string | RegExp> }}
+ */
+export function normalizeMiddlewareEntry(
+  entry: any
```

Comment:
If we don't have a more specific type for `entry`, `unknown` is better than `any`.

```suggestion
  entry: unknown
```